# <h2>Importación de librerias</h2>

In [3]:
# Visualization Libraries
import matplotlib.pyplot as plt
import plotly.express as px
import seaborn as sns

# Operational Libraries
import pandas as pd
import numpy as np

# <h2>1. Carga de datos desde excel a dataframes de pandas</h2>


In [8]:
file_id = '10xZY27WYUZ-wH1Aewqyt4eFf6oh75LJA'
url = f'https://drive.google.com/uc?id={file_id}'
df = pd.read_csv(url)


In [9]:
df

Unnamed: 0     rut     Genero   Categoria_Digital Elasticidad_Precios  \
0                0       1  Masculino  Cliente no Digital                Alta   
1                1       2   Femenino  Cliente no Digital                Baja   
2                2       3   Femenino  Cliente no Digital                Baja   
3                3       4  Masculino  Cliente no Digital                Alta   
4                4       5  Masculino     Cliente Digital               Media   
...            ...     ...        ...                 ...                 ...   
543646      543646  543647   Femenino     Cliente Digital                Baja   
543647      543647  543648  Masculino  Cliente no Digital                Baja   
543648      543648  543649  Masculino  Cliente no Digital               Media   
543649      543649  543650   Femenino  Cliente no Digital               Media   
543650      543650  543651   Femenino  Cliente no Digital               Media   

       Nacionalidad  Propension  Probabilidad_No_Pago  Edad         Renta  \
0           Chileno    0.997340              0.028445  30.0  6.258183e+05   
1           Chileno    0.291601              0.014320  41.0  3.172616e+05   
2           Chileno    0.685085              0.002156  38.0  1.240551e+07   
3           Chileno    0.914672              0.034418  57.0  5.441466e+05   
4           Chileno    0.425077              0.014978  26.0  1.870225e+05   
...             ...         ...                   ...   ...           ...   
543646      Chileno    0.144657              0.037291  29.0  1.176598e+05   
543647      Chileno    0.740170              0.035877  31.0  1.558612e+06   
543648      Chileno    0.255285              0.023306  49.0  9.449508e+05   
543649      Chileno    0.709086              0.015121  40.0  1.039964e+06   
543650      Chileno    0.860781              0.019647  35.0  4.728067e+05   

        Oferta_Consumo     Deuda_CMF  Tiempo_como_cliente  
0            2164276.0  7.125854e+05                    2  
1            1451704.0  1.970697e+05                    2  
2           39322607.0  1.400604e+07                    2  
3            1607042.0  5.159465e+05                    2  
4             493865.0  2.601013e+05                    2  
...                ...           ...                  ...  
543646       9625991.0  1.299826e+05                    2  
543647       7522126.0  1.740917e+06                    2  
543648       1889172.0  1.057952e+06                    2  
543649       4518498.0  1.010872e+06                    2  
543650       1979540.0  5.745756e+05                    2  

[543651 rows x 13 columns]

<h3>1.1 Carga de 'Informacion_Clientes.csv'</h3>

En primer lugar se cargará la información de los clientes. Esto incluye las siguientes características de los clientes:

* **unnamed**: algo como uid
* **Rut**: identificador de Chile (supongo que por privacidad va desde 0 a max de observaciones)
* **Género**: Masculino o femenino
* **Categoría_Digital**: Si el cliente es digital o no
* **Elasticidad_Precios**: Baja, media o alta
* **Nacionalidad**: Chileno o extranjero
* **Propensión**: Número entre 0 y 1 que idica que tan propenso a cursar un credito es el cliente
* **Probabilidad_No_Pago**: Número entre 0 y 1 que indica la probabilidad de que el cliente no pague la deuda
* **Edad**: Numero entero de edad en años
* **Renta**: Renta promedio de los últimos 12 meses
* **Oferta_Consumo**: Monto máximo que puede cursar un cliente dado sus antecedentes crediticios y situación socioeconómica. 
* **Deuda_CMF**: Deuda que tiene el cliente en otros bancos. Efectivamente es deuda pendiente, pero de créditos otorgados por la competencia.
* **Tiempo_como_cliente**: Número de tiempo(no sé en que medida está) que el cliente lleva en el banco
